entproto/internal/entprototest/service_test.go

```diff
@@ -14,30 +14,140 @@
+
+	methodUpdate = method{
+		name:         "Update",
+		input:        "Update%sRequest",
+		output:       "%s",
+		formatInput:  true,
+		formatOutput: true,
+	}
+
+	methodDelete = method{
+		name:         "Delete",
+		input:        "Delete%sRequest",
+		output:       "Empty",
+		formatInput:  true,
+		formatOutput: false,
+	}
+)
+
 func (suite *AdapterTestSuite) TestServiceGeneration() {
-	fd, err := suite.adapter.GetFileDescriptor("BlogPost")
-	suite.Require().NoError(err)
-
-	svc := fd.FindService("entpb.BlogPostService")
-	suite.NotNil(svc)
-
-	getMeth := svc.FindMethodByName("Get")
-	suite.Require().NotNil(getMeth)
-	suite.EqualValues("GetBlogPostRequest", getMeth.GetInputType().GetName())
-	suite.EqualValues("BlogPost", getMeth.GetOutputType().GetName())
-
-	createMeth := svc.FindMethodByName("Create")
-	suite.Require().NotNil(createMeth)
-	suite.EqualValues("CreateBlogPostRequest", createMeth.GetInputType().GetName())
-	suite.EqualValues("BlogPost", createMeth.GetOutputType().GetName())
-
-	deleteMeth := svc.FindMethodByName("Delete")
-	suite.Require().NotNil(deleteMeth)
-	suite.EqualValues("DeleteBlogPostRequest", deleteMeth.GetInputType().GetName())
-	suite.EqualValues("google.protobuf.Empty", deleteMeth.GetOutputType().GetFullyQualifiedName())
-
-	updateMeth := svc.FindMethodByName("Update")
-	suite.Require().NotNil(updateMeth)
-	suite.EqualValues("UpdateBlogPostRequest", updateMeth.GetInputType().GetName())
-	suite.EqualValues("BlogPost", updateMeth.GetOutputType().GetName())
+	testCases := []struct {
+		testName        string
+		schemaName      string
+		includedMethods []method
+		excludedMethods []method
+	}{
+		{
+			testName:   "Default Method Generation",
+			schemaName: "BlogPost",
+			includedMethods: []method{
+				methodCreate,
+				methodGet,
+				methodUpdate,
+				methodDelete,
+			},
+		},
+		{
+			testName:   "All Methods Generation",
+			schemaName: "AllMethodsService",
+			includedMethods: []method{
+				methodCreate,
+				methodGet,
+				methodUpdate,
+				methodDelete,
+			},
+		},
+		{
+			testName:   "One Method Generation",
+			schemaName: "OneMethodService",
+			includedMethods: []method{
+				methodGet,
+			},
+			excludedMethods: []method{
+				methodCreate,
+				methodUpdate,
+				methodDelete,
+			},
+		},
+		{
+			testName:   "Two Method Generation",
+			schemaName: "TwoMethodService",
+			includedMethods: []method{
+				methodCreate,
+				methodGet,
+			},
+			excludedMethods: []method{
+				methodUpdate,
+				methodDelete,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		println(fmt.Sprintf("Test %s", tc.testName))
+
+		fd, err := suite.adapter.GetFileDescriptor(tc.schemaName)
+		suite.Require().NoError(err)
+
+		svc := fd.FindService(fmt.Sprintf("entpb.%sService", tc.schemaName))
+		suite.NotNil(svc)
+
+		for _, m := range tc.includedMethods {
+			getMeth := svc.FindMethodByName(m.name)
+			suite.Require().NotNil(getMeth)
+
+			if m.formatInput {
+				m.input = m.getFormatInput(tc.schemaName)
+			}
```

Comment:
sorry i dont understand, what is this doing? seems like formatInput is always true?

Reply:
The idea was to leave this here for future methods that might have an input message that isn't based on the schema name. (Like google.protobuf.Empty)